src/index.js

```diff
@@ -55,40 +61,48 @@ exports.compare = memoize(function (versionA, versionB) {
     if (versionB === null) {
         return 1;
     }
-    versionA = normalize(versionA);
-    versionB = normalize(versionB);
-    var isAValid = semver.valid(versionA);
-    var isBValid = semver.valid(versionB);
+    var nVersionA = normalize(versionA);
+    var nVersionB = normalize(versionB);
+    var isAValid = semver.valid(nVersionA);
+    var isBValid = semver.valid(nVersionB);
     if (isAValid && !isBValid) {
         return 1;
     }
     if (!isAValid && isBValid) {
         return -1;
     }
     if (!isAValid && !isBValid) {
-        if (versionA > versionB) {
+        if (nVersionA > nVersionB) {
             return 1;
         }
-        if (versionA < versionB) {
+        if (nVersionA < nVersionB) {
             return -1;
         }
         return 0;
     }
-    var semverResult = semver.compare(versionA, versionB);
+    var semverResult = semver.compare(nVersionA, nVersionB);
     if (semverResult !== 0) {
         return semverResult;
     }
-    var revA = getRev(versionA);
-    var revB = getRev(versionB);
-    if (revA !== revB) {
-        return revA > revB ? 1 : -1;
-    }
-    var devA = Number(isDevelopmentVersion(versionA));
-    var devB = Number(isDevelopmentVersion(versionB));
-    if (devA !== devB) {
-        return devA > devB ? -1 : 1;
+    var comparisonMethods = [
+        { fn: getRev, desc: false, normalize: true },
+        { fn: isPreReleaseVariant, desc: true, normalize: true },
+        { fn: getPreReleaseVariant, desc: false, normalize: true },
+        { fn: getPostReleaseVariant, desc: false, normalize: false },
```

Comment:
This reminds me the simplicity of
`ORDER BY rev, isPreRelease DESC, preReleaseVariant, postReleaseVariant`